openapi/components/schemas/StorefrontUpcomingInvoice.yaml

```
@@ -0,0 +1,113 @@
+type: object
+readOnly: true
+properties:
+  id:
+    description: ID of the upcoming invoice, which is persisted in the future renewal invoice.
+    type: string
+    readOnly: true
+    maxLength: 50
+    example: in_0YVF9605RKC62BP14NE2R7V2XT
+  websiteId:
+    $ref: ./WebsiteId.yaml
+  subscriptionId:
+    description: ID of the related subscription order.
+    type: string
+    readOnly: true
+    maxLength: 50
+    example: ord_01GYJPRKHBD6ZYHH897QCJMBS4
+  currency:
+    $ref: ./CurrencyCode.yaml
+  items:
+    type: array
+    description: Upcoming invoice items array.
+    items:
+      $ref: ./StorefrontUpcomingInvoiceItem.yaml
```

```suggestion
      $ref: ./UpcomingInvoiceItem.yaml
```

Perhaps to the same way as for StorefrontInvoice and use existing schema for items.

---

Changed. Thanks